src/components/PupilComponents/LessonEngineProvider/LessonEngineProvider.tsx

```diff
@@ -177,16 +183,32 @@ export const useLessonEngineContext = () => {
 export type LessonEngineProviderProps = {
   children: ReactNode;
   initialLessonReviewSections: Readonly<LessonReviewSection[]>;
+  initialSection: LessonSection;
 };
 
 export const LessonEngineProvider = memo(
-  ({ children, initialLessonReviewSections }: LessonEngineProviderProps) => {
+  ({
+    children,
+    initialLessonReviewSections,
+    initialSection,
+  }: LessonEngineProviderProps) => {
     const [state, dispatch] = useReducer(lessonEngineReducer, {
       lessonReviewSections: initialLessonReviewSections,
-      currentSection: "overview",
+      currentSection: initialSection,
       lessonStarted: false,
       sections: {},
     });
+    const navigateToSection = useNavigateToSection();
+
+    // Sync the currentSection when the user navigates back/forward
+    useLessonPopStateHandler((section) => {
+      dispatch({ type: "setCurrentSection", section });
+    });
+
+    // Update the URL when `currentSection` changes
+    useEffect(() => {
+      navigateToSection(state.currentSection);
```

Comment:
These sync changes to the current section with the address bar and create the corresponding entires in the browser history

Reply:
UseEffect (raises eyebrow)

Reply:
![mister-gotcha-4-9faefa-1](https://github.com/oaknational/Oak-Web-Application/assets/122096/bd815249-d69e-4bfe-b841-1f7c2a9b7e0b)